pkg/plugins/runtime/k8s/controllers/pod_controller.go

```diff
@@ -352,6 +366,36 @@ func (r *PodReconciler) createOrUpdateDataplane(
 	return nil
 }
 
+func (r *PodReconciler) hasResourceChange(ctx context.Context, resource kube_client.Object, newResourceSpec model.ResourceSpec) (bool, error) {
+	key := kube_client.ObjectKeyFromObject(resource)
+	// This call shouldn't affect performance because if we don't call it here `createOrUpdate` does this call.
+	// If the resource change, `createOrUpdate` calls it 2nd time but in this case it should use the value
+	// from the cache.
+	if err := r.Client.Get(ctx, key, resource); err != nil {
+		if !kube_apierrs.IsNotFound(err) {
+			return false, err
+		}
+	}
```

Comment:
Maybe we can just return as if `NotFound` occurred which could avoid the later `model.Equal` to be called to save performance cost.

```text
	if err := r.Client.Get(ctx, key, resource); err != nil {
		if kube_apierrs.IsNotFound(err) {
		        log.debug(...)
			return false, nil
		}else{
		   return false, err
		}
	}
```